lib/types.js

```diff
@@ -557,47 +557,54 @@ module.exports = function (expect) {
-                                        this.shouldEqualError(diffItem.expected, inspect);
-                                    });
-                                }
-                            }), true);
-                        });
-                    }
-                });
-            });
+                                .amend(delimiterOutput.sp()).error('// should be moved');
+                        } else if (diffItem.type === 'insert') {
+                            this.annotationBlock(function () {
+                                this.error('missing ').block(function () {
+                                    var index = typeof diffItem.actualIndex !== 'undefined' ? diffItem.actualIndex : diffItem.expectedIndex;
+                                    this.property(index, inspect(diffItem.value), true);
+                                });
+                            });
+                        } else if (diffItem.type === 'remove') {
+                            this.block(function () {
+                                this.property(diffItem.actualIndex, inspect(diffItem.value), true)
+                                    .amend(delimiterOutput.sp()).error('// should be removed');
+                            });
+                        } else if (diffItem.type === 'equal') {
+                            this.block(function () {
+                                this.property(diffItem.actualIndex, inspect(diffItem.value), true)
+                                    .amend(delimiterOutput);
+                            });
+                        } else {
+                            this.block(function () {
+                                var valueDiff = diff(diffItem.value, diffItem.expected);
+                                this.property(diffItem.actualIndex, output.clone().block(function () {
+                                    if (valueDiff && valueDiff.inline) {
+                                        this.append(valueDiff.amend(delimiterOutput));
+                                    } else if (valueDiff) {
+                                        this.append(inspect(diffItem.value).amend(delimiterOutput.sp())).annotationBlock(function () {
+                                            this.shouldEqualError(diffItem.expected, inspect).nl(2).append(valueDiff);
+                                        });
+                                    } else {
+                                        this.append(inspect(diffItem.value).amend(delimiterOutput.sp())).annotationBlock(function () {
+                                            this.shouldEqualError(diffItem.expected, inspect);
+                                        });
+                                    }
+                                }), true);
+                            });
+                        }
+                    });
+                }
+            }));
```

Comment:
Yeah, a long time ago we made the decision to keep the array/object diffs separate from the generation of the 'to satisfy' diffs. Since then there has been a lot of fluctuation back and forth, but I think they're still sufficiently different that it could pose a challenge to reconsolidate them.

If we decide to go for https://github.com/unexpectedjs/unexpected/issues/330 they would align closer.